model/ppis.js

```diff
@@ -9,74 +9,76 @@ Annotations = require('./annotations'),
 }
 
 exports.upsertPPI = function(data, callback) {
-    ppis = Schemas.interactions
-    annos = Schemas.annotations
-
-    // insert into the annos
-    annoInsertQuery = annos.insert([{
-	user_id : data.user_id,
-	reference : data.pmid,
-	ref_source: data.anno_source,
-	comment : data.comment,
-	type : "ppi" }]).returning(annos.u_id)
-
-    handleErr = function(err, subresult, query) {
-	if (err == null && subresult.rows.length == 0) {
-	    err = Error("Did not return ppi annotation ID")
-	}
-	if (err) {
-            console.log("Error upserting ppi annotation: " + err);
-	    console.log("Debug: full query:", query.toQuery().text)
-	    console.log("Debug: full query:", query.toQuery().values)
-	    callback(err, null)
-	}
+  var ppis = Schemas.interactions,
+      annos = Schemas.interaction_annotations;
+
+  // insert into the annos
+  annoInsertQuery = annos.insert(
+    annos.user_id.value(data.user_id),
+    annos.reference.value(data.pmid),
+    annos.ref_source.value(data.anno_source),
+    annos.comment.value(data.comment),
+    annos.type.value('ppi')).returning(annos.u_id)
+
+  handleErr = function(err, subresult, query) {
+    if (err == null && subresult.rows.length == 0) {
+      err = Error("Did not return ppi annotation ID")
+    }
+    if (err) {
+      console.log("Error upserting ppi annotation: " + err);
+      console.log("Debug: full query:", query.toQuery().text)
+      console.log("Debug: full query:", query.toQuery().values)
+      callback(err, null)
     }
+  }
 
-//    console.log("Submitting upsert query: ", annoInsertQuery.toQuery().text)
 
     // todo: test update case
     // todo: transaction-ize w/ rollback (not necessary, just good to clean the annos table)
-    Database.execute(annoInsertQuery, function(err, subresult) {
-	// retrieve the unique ID for the annotation
-	handleErr(err, subresult, annoInsertQuery)
-	u_id = subresult.rows[0].u_id
-
-	// database, type, weight, directed, tissue are all unspecified on front-end
-	ppiInsertQuery = ppis.insert(
-	    ppis.source.value(data.source),
+  Database.execute(annoInsertQuery, function(err, subresult) {
+    // retrieve the unique ID for the annotation
+    handleErr(err, subresult, annoInsertQuery)
+    u_id = subresult.rows[0].u_id
+
+    // database, type, weight, directed, tissue are all unspecified on front-end
+    ppiInsertQuery = ppis.insert(
+      ppis.source.value(data.source),
 	    ppis.target.value(data.target),
 	    ppis.weight.value(data.weight),
 	    ppis.database.value(data.database),
 	    ppis.anno_id.value(u_id)).returning(ppis.anno_id) // we can return more if we want
 
-	Database.execute(ppiInsertQuery, function(err, result) {
-	    handleErr(err, result, ppiInsertQuery)
-	    callback(null, result.rows[0])
-	})
-    })
+    Database.execute(ppiInsertQuery, function(err, result) {
+      handleErr(err, result, ppiInsertQuery)
+      callback(null, result.rows[0])
+    });
+  });
 }
 
 // Loads annotations into the database
```

Comment:
Do we need to load annotations from file in MAGI anymore? Aren't they all in Postgres?